rx_signal_processing/dsp.py

```diff
@@ -182,13 +197,17 @@ def beamform_samples(self, filtered_samples, beam_phases):
         :type       beam_phases:       list
 
         """
-        beam_phases = xp.array(beam_phases)
-
-        # [num_slices, num_antennas, num_samples]
         # [num_slices, num_beams, num_antennas]
-        beamformed_samples = xp.einsum('ijk,ilj->ilk', filtered_samples, beam_phases)
+        beam_phases = np.array(beam_phases)
 
-        self.beamformed_samples = beamformed_samples
+        # [num_slices, num_beams, num_samples]
+        final_shape = (filtered_samples.shape[0], beam_phases.shape[2], filtered_samples.shape[2])
+        final_size = np.dtype(np.complex64).itemsize * filtered_samples.shape[0] * filtered_samples.shape[2] * beam_phases.shape[2]
+        bf_shm = shared_memory.SharedMemory(create=True, size=final_size)
+        self.beamformed_samples = np.ndarray(final_shape, dtype=np.complex64, buffer=bf_shm.buf)
+        self.beamformed_samples = np.einsum('ijk,ilj->ilk', filtered_samples, beam_phases)
+
```

Comment:
My thought was that in case it ever needed to change, then you would only need to change it in one place (actually having it in the config file might be most general - but probably overkill). It's prob fine the way it is - I can't imagine it changing much in the future.